cloudinit/sources/azure/imds.py

```diff
@@ -46,7 +49,10 @@ def exception_callback(self, req_args, exception) -> bool:
             return False
 
         log = True
-        retry = True
+        if self.retry_timeout is not None and time() >= self.retry_timeout:
```

Comment:
Won't this always return True? `time()` returns seconds since epoch, but `self.retry_timeout` will be something like 300.

Reply:
I don't think so? If you look at the source of retry_timeout in get_metadata_from_imds(), it's configured to be `retry_timeout = time() + 300`